simpleclient/src/main/java/io/prometheus/client/Collector.java

```diff
@@ -396,4 +394,17 @@ public static String doubleToGoString(double d) {
     }
     return Double.toString(d);
   }
+
+  protected static final String DISABLE_CREATED_SERIES = "PROMETHEUS_DISABLE_CREATED_SERIES";
+  private static final List<String> TRUTHS = Arrays.asList("true", "1", "t");
+
+  protected static boolean USE_CREATED = getUseCreated();
```

Comment:
* Please make it `final`. It's not supposed to change.
* If you make it `final` but still want to test it, you should not make it `static`.
* Rename to `CREATED_SERIES_DISABLED` because that makes it clearer that this is related to the `PROMETHEUS_DISABLE_CREATED_SERIES` environment variable.
* If you revert the changes in `getNames()`, you could move this one level down in the class hierarchy to `SimpleCollector`.

Reply:
> * Please make it `final`. It's not supposed to change.
> * If you make it `final` but still want to test it, you should not make it `static`.
Yes, it was final - but then I could not find a way to test it. It I remove the `static` every object will have its own copy of it which seems to be excessive. The above is a compromise.

> * Rename to `CREATED_SERIES_DISABLED` because that makes it clearer that this is related to the `PROMETHEUS_DISABLE_CREATED_SERIES` environment variable.
That will flip it's meaning and will change all code from ```
      if (USE_CREATED) {
```
to
```
      if (!CREATED_SERIES_DISABLED) {
```
Which is a double negative and is considered as a poor programming practice. For example - https://cleankotlin.nl/blog/double-negations#:~:text=Double%20negations%20are%20not%20not,can%20hide%20in%20your%20program.

> * If you revert the changes to `getNames()`, you could move this one level down in the class hierarchy to `SimpleCollector`.
Not sure of the benefit - let's resolve `getNames()` discussion first.

Reply:
I think if the flag should not change at runtime it should be `final`, even if that makes it a bit harder to test. There are a few options how to write a test if it's `final`:
* Make it non-static
* Set the field via reflection (you can remove the `final` modifier via reflection)
* Write an integration test with Testcontainers test (then you could set the environment variable before starting the test application).

Regarding the name of the flag: It's always hard to maintain code if there are different names for the same concept. If we name the environment variable `PROMETHEUS_DISABLE_CREATED_SERIES` but the flag `USE_CREATED` somebody reading the code would always need to double-check if these mean the same thing. If we use a similar name, it's more obvious.

Reply:
> I think if the flag should not change at runtime it should be `final`, even if that makes it a bit harder to test. There are a few options how to write a test if it's `final`:
> 
> * Make it non-static
> * Set the field via reflection (you can remove the `final` modifier via reflection)
> * Write an integration test with Testcontainers test (then you could set the environment variable before starting the test application).

I'm not sure what I can add. The extra variable per object seems like too much a price to pay given the alternative. 

Can you please provide some guidance on how to use `Testcontainers` here? I'm unfamiliar with that framework.
Otherwise it seems like a good way on making the variable final (which something I wanted to do in the first place). It will also make the new dependency redundant.

> Regarding the name of the flag: It's always hard to maintain code if there are different names for the same concept. If we name the environment variable `PROMETHEUS_DISABLE_CREATED_SERIES` but the flag `USE_CREATED` somebody reading the code would always need to double-check if these mean the same thing. If we use a similar name, it's more obvious.

Thanks for pointing this rule out. We're talking about readability multiple places vs readability one place. For me the many wins - the condition should be read as "TRUE" instead of "NOT NOT". Finally, the this choice mirrors the python one. The environment variable is mentioned once in a function and is pretty easy to lookup in any decent IDE.